### ISSUE TITLE
Matchmaker rating shown as 0

### ISSUE THREAD

**Issue by BlackYps** [FAForever/downlords-faf-client#2480]
The client shows no 4v4 matchmaker rating for many people when they play their first game even though the initialization happens correctly on the server. So it defaults to 0 in game.
Oddly enough for some people it works.
Correct rating initialization can be confirmed by looking at the ratings of the replay after the game has finished

**Comment by Sheikah45** [comment]
This happens because the server does the initialization internally and doesn't send the client the players rating before the game starts so the client defaults to zero. So this isn't really a client issue because it doesn't have the necessary information

**Comment by BlackYps** [comment]
Then why does it work for some players?

**Comment by Sheikah45** [comment]
Largely this is probably just a race condition which is why it works for some people

**Comment by BlackYps** [comment]
Also, @Askaholic says they are in the player_info message, so the client can get them from there. I suspect that is why it works for some people, depending on when the message arrives

**Comment by Sheikah45** [comment]
The player_info message is not sent out right as soon as a change occurs. The server batches up the players and sends then in intervals. So that is the race condition. If it gets an updated message before the game starts it will show the initialized rating. If it doesn't then the players will show as 0.

**Comment by Askaholic** [comment]
Makes sense. Probably the right thing to do would be to add the rating to the `game_launch` message?

**Comment by Sheikah45** [comment]
for who all the players just the player who the message is going to? Also I will move this to the server
